flaml/ml.py

```diff
@@ -75,6 +104,51 @@ def get_estimator_class(task, estimator_name):
     return estimator_class
 
 
+def metric_loss_score(
+        metric_name,
+        y_predict,
+        y_true,
+        labels=None,
+        sample_weight=None,
+        groups=None,
+    ):
+    import datasets
+    if metric_name in sklearn_metric_name_set:
+        return sklearn_metric_loss_score(metric_name, y_predict, y_true, labels, sample_weight, groups)
+    else:
+        try:
+            """
+                hf's datasets.load_metric("pearsonr") returns nan (hf's bug), overwriting it here
+            """
+            if metric_name == "spearmanr":
+                from scipy.stats import spearmanr
+                y_true = y_true.to_list() if type(y_true) == pd.Series else list(y_true)
+                score = spearmanr(list(y_predict), y_true)[0]
+                metric_mode = "max"
+            elif metric_name == "pearsonr":
+                from scipy.stats import pearsonr
+                y_true = y_true.to_list() if type(y_true) == pd.Series else list(y_true)
+                score = pearsonr(list(y_predict), y_true)[0]
+                metric_mode = "max"
+            else:
+                metric = datasets.load_metric(metric_name)
+                metric_mode = huggingface_metric_to_mode[metric_name]
+                score = metric.compute(predictions=y_predict, references=y_true)[
+                                    metric_name
+                                ]
+            multiplier = -1 if metric_mode == "max" else 1
+            return score * multiplier
+        except FileNotFoundError:

```

Comment:
Why expecting this error?

Reply:
Added comments

Reply:
if that exception only happens in load_metric(), the try...except needs to only contain that part.